test/mlir/accelerators/nnpa/conversion/onnx-to-zhigh/pool.mlir

```diff
@@ -101,12 +101,26 @@ func @averagepool_should_lower_to_zhigh_same_upper(%arg0: tensor<1x3x32x32xf32>)
 }
 
 // -----
-// CHECK-LABEL:  func @test_pool_not_lowered_3d
-func @test_pool_not_lowered_3d(%arg0: tensor<1x3x32x32x32xf32>) -> (tensor<1x3x31x31x31xf32>, tensor<1x3x31x31x31xf32>) {
-  %0 = "onnx.MaxPoolSingleOut"(%arg0) {auto_pad = "NOTSET", dilations = [1, 1, 1], kernel_shape = [2, 2, 2], pads = [0, 0, 0, 0, 0, 0], strides = [1, 1, 1]} : (tensor<1x3x32x32x32xf32>) -> tensor<1x3x31x31x31xf32>
+// COM: Pooling in zDNN only support 4D input (MaxPool1D and MaxPool3D not suported)
+// CHECK-LABEL:  func @test_pool_not_lowered_input3d
+func @test_pool_not_lowered_input3d(%arg0: tensor<1x3x32xf32>) -> (tensor<1x3x31xf32>, tensor<1x3x31xf32>) {
+  %0 = "onnx.MaxPoolSingleOut"(%arg0) {kernel_shape = [2]} : (tensor<1x3x32xf32>) -> tensor<1x3x31xf32>
   // CHECK: "onnx.MaxPoolSingleOut"
 
-  %1 = "onnx.AveragePool"(%arg0) {auto_pad = "NOTSET", kernel_shape = [2, 2, 2], pads = [0, 0, 0, 0, 0, 0], strides = [1, 1, 1]} : (tensor<1x3x32x32x32xf32>) -> tensor<1x3x31x31x31xf32>
+  %1 = "onnx.AveragePool"(%arg0) {kernel_shape = [2]} : (tensor<1x3x32xf32>) -> tensor<1x3x31xf32>
+  // CHECK: "onnx.AveragePool"
+
+  return %0, %1 : tensor<1x3x31xf32>, tensor<1x3x31xf32>
+}
+
+// -----
+// COM: Pooling in zDNN only support 4D input (MaxPool1D and MaxPool3D not suported)
+// CHECK-LABEL:  func @test_pool_not_lowered_input5d
```

Comment:
The above line said `MaxPool1D and MaxPool3D`, and you had `test_pool_not_lowered_input3d` for 3D, so this should be `test_pool_not_lowered_input1d` instead of `test_pool_not_lowered_input5d`? It seems they are inconsistent.

Reply:
Sorry for confusing. Legality check for MaxPool checks if dimension size of input is 4 since input for MaxPool2D need to have 4d (NCHW). Input is 3d in MaxPool1D (NCL), and 5d in MaxPool3D(NCDHW) .    
So, I named the functions as follows.
- test_pool_not_lowered_input3d:  ==> not lowered when input tensor is 3d (MaxPool1D case)
- test_pool_not_lowered_input5d:  ==> not lowered when input tensor is 5d (MaxPool3D case)
- Only lowered when input tensor is 4d (MaxPool2D case)

However, simply `test_pool_not_lowered_maxpool1d` and `test_pool_not_lowered_maxpool3d` might be better.

Reply:
OK. I see. Thanks!